fog/view/server/src/server.rs

```diff
@@ -369,7 +366,8 @@ where
                 WorkerTickResult::HasMoreWork => {}
 
                 WorkerTickResult::Sleep => {
-                    sleep(DB_POLL_INTERNAL);
+                    // This waits between polling. Is this doubled up with sleeps in DbFetcher?
```

Comment:
It's not doubled up, the db fetcher looks to be launched on a separate thread.

I think this thread could be running at a faster rate with no extra load on the db as it is pulling form the db fetcher and storing in the enclave
Polling faster here won't move faster than the 250ms from the db_fetcher, but it would reduce the overall latency. Right now worst case latency is 250ms(db fetcher) + 250ms (this) to get the data into the enclave.

The faster this moves the closer the worst case latency get's to 250ms

Reply:
Should we set the worker tick sleep to something less than the db polling interval?  I just assumed that it should be the same because they shared the same const. 

I think the down side to reducing the worker sleep, is that we could end up in a situation where the view server reports ready prematurely. 

IIUC It looks like worker loop just sets the service to ready when the fetcher stops returning records.  Even if the interval timings are the same this could be problematic if the time it takes to fetch records from the db exceeds the interval time + processing time the worker takes to get its tasks done.  Could explain why view gets pretty unhappy in high network latency scenarios. 

This also seems like it could be a place where we could put logic in to control database polling based on current processed block height.  Can we shutdown the dbFetcher thread from here?

Reply:
I think for now we can keep the tick the same and see what happens.
From the comment here, [MAX_QUEUED_RECORDS](https://github.com/mobilecoinfoundation/mobilecoin/blob/8acb950f86af0ca2cfc8424ab18252ec3c4a0b66/fog/view/server/src/db_fetcher.rs#L30) I think the worker will be the slower one

> Can we shutdown the dbFetcher thread from here?
I didn't think so, but it looks like view is using `max` when merging the shard data, [LastKnownData](https://github.com/mobilecoinfoundation/mobilecoin/blob/8acb950f86af0ca2cfc8424ab18252ec3c4a0b66/fog/view/enclave/impl/src/types.rs#L167)
Ledger was using `min()` so it broke things. We maybe able to turn off fetcher once we're done.